### PR TITLE
Search Function

### PR DESCRIPTION
**Pull request by lfiolhais**
This PR adds a search command to OMF per #292. The search command supports searches for both themes and packages. To execute a search run `omf search` with the flags `-t / --theme` for themes or `-pkg / --package` for packages. If neither flag is present both themes and packages will be searched.

The output is as follow:
```
↪ omf search t
Packages
apt				python
artisan				rustup
await				ssh-term-helper
basename-compat			tab
battery				technicolor
export				termux
extract				thefuck
fonts				tiny
getopts				title
git-flow			tmux-zen
homebrew-command-not-found	weather
kill-on-port

Themes
agnoster		integral		taktoa
batman			jacaetevha		technopagan
bobthefish		mtahmed			toaster
default			numist			tomita
gitstatus		pastfish		trout
godfather		simple-ass-prompt

↪ omf search -t simple
Themes
simple-ass-prompt	simplevi

↪ omf search -pkg nv
Packages
direnv		foreign-env	nvm		rbenv
```

This my first "big" PR to OMF so any feedback is welcomed and appreciated. :smile:

### CODE REVIEW

File: pkg/omf/functions/search/omf.search.fish [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      switch "$argv[1]"
+        case "-pkg" "--package";
+          __omf.search.pkg $argv[2]
+        case "-t" "--theme";
+          __omf.search.theme $argv[2]
+        case '*';
+          __omf.search.usage
+      end
+    case '*';
+      __omf.search.usage
```
Comment (derekstavis) [inline pkg/omf/functions/search/omf.search.fish]: The idea is that all command line interface stuff like parameter should be done in `omf.cli.*` functions. Please move this "0 arguments" case there :+1:

**Comment by derekstavis** [comment]
Absolutely loved it! Very cool! I will take a closer look at weekend!

**Comment by lfiolhais** [comment]
I've update the code according to your feedback.

**Comment by coderstephen** [comment]
I can test this some more later today, and then let's see this get merged! Great work, @lfiolhais!

**Comment by coderstephen** [comment]
@lfiolhais I see just one bug with `omf.cli.search`; other than that, it works excellently. Fix the bug and I think we can merge!

File: pkg/omf/functions/cli/omf.cli.search.fish [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      echo
+      omf.search.theme $argv[1]
+    case 2;
+      switch "$argv[1]"
+        case "-pkg" "--package";
+          omf.search.pkg $argv[2]
+        case "-t" "--theme";
+          omf.search.theme $argv[2]
+        case '*';
+          __omf.search.usage
```
Comment (coderstephen) [inline pkg/omf/functions/cli/omf.cli.search.fish]: This is a typo I think; it should be `__omf.cli.search.usage` instead of `__omf.search.usage`.

**Comment by lfiolhais** [comment]
@coderstephen Fixed!

**Comment by coderstephen** [comment]
@lfiolhais Excellent; everything looks good to me. Just rebase onto master and then I can merge. Thanks for the contribution!

**Comment by lfiolhais** [comment]
The error reported by travis is unrelated to my changes.

**Comment by coderstephen** [comment]
:+1:

**Pull request merged**